pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -673,6 +673,13 @@ type Module struct {
 	// Enabled determines whether a module should be enabled or not
 	// +optional
 	Enabled bool `json:"enabled,omitempty"`
+	// Settings to further configure the module
+	Settings ModuleSettings `json:"settings,omitempty"`
+}
+
+type ModuleSettings struct {
```

Comment:
From an API perspective it is a bit odd that a module could define settings that aren't related to it. But it is nice to keep it simple and strongly type the settings, so it seems ok since we would just ignore them if applied to the wrong module.